File: content/problems/0228-summary-ranges/solutions/run-scanning/solution.cpp
class Solution {
    public: vector<string> summaryRanges(vector<int>& nums){
        vector<string> answer;
        for(int start=0;start<(int)nums.size();){
            int end=start;
            while(end+1<(int)nums.size()&&(long long)nums[end+1]-nums[end]==1)++end;
            answer.push_back(start==end?to_string(nums[start]):to_string(nums[start])+"->"+to_string(nums[end]));
            start=end+1;
        }
        return answer;
    }
};

File: content/problems/0228-summary-ranges/solutions/run-scanning/Solution.kt
class Solution {
    fun summaryRanges(nums:IntArray):List<String>{
        val answer=mutableListOf<String>();
        var start=0;
        while(start<nums.size){
            var end=start;
            while(end+1<nums.size&&nums[end+1].toLong()-nums[end]==1L)end++;
            answer.add(if(start==end)nums[start].toString() else "${nums[start]}->${nums[end]}");
            start=end+1
        };
        return answer
    }
}
